File: Programming-languages-and-laboratory/Java/Exercises/26_compito_java_2019_11_19/GestioneConferenze.java
import java.io.*;
import java.util.*;

public class GestioneConferenze {

	static List<Partecipante> partecipanti = new LinkedList<Partecipante>();
	static List<Conferenza> conferenze = new LinkedList<Conferenza>();
	static Map<Integer, Partecipante> codPart = new HashMap<Integer, Partecipante>();

	public static void main(String[] args) {

		// PUNTO 1: memorizzazione progetti
		try {
			BufferedReader br = new BufferedReader(new FileReader("partecipanti.txt"));
			String line = br.readLine();
			while (line != null) {
				StringTokenizer tok = new StringTokenizer(line);
				int cod = Integer.parseInt(tok.nextToken());
				String tipo = tok.nextToken();
				String nome = br.readLine();
				String cognome = br.readLine();

				if (tipo.equals("studente")) {
					String corso = br.readLine();
					String uni = br.readLine();
					line = br.readLine();
					int anno = Integer.parseInt(line);
					Studente s = new Studente(cod, nome, cognome, corso, uni, anno);
					partecipanti.add(s);
					codPart.put(cod, s);
				} else {
					String ente = br.readLine();
					line = br.readLine();
					tok = new StringTokenizer(line);
					int dip = Integer.parseInt(tok.nextToken());
					boolean acc = Boolean.parseBoolean(tok.nextToken());
					Regolare r = new Regolare(cod, nome, cognome, ente, dip, acc);
					partecipanti.add(r);
					codPart.put(cod, r);
				}
				line = br.readLine();
			}
			br.close();
		} catch (IOException e) {
			System.err.println(e);
		} catch (Exception e) {
			System.err.println(e);
		}
		// PUNTO 2: lettura di tutti i ricercatori

		try {
			BufferedReader br = new BufferedReader(new FileReader("conferenze.txt"));
			String line = br.readLine();
			while (line != null) {
				String nome = line;
				String luogo = br.readLine();
				Conferenza c = new Conferenza(nome, luogo);
				conferenze.add(c);
				line = br.readLine();
				while (!line.equals("")) {
					StringTokenizer tok = new StringTokenizer(line);
					int cod = Integer.parseInt(tok.nextToken());
					int reg = Integer.parseInt(tok.nextToken());
					Partecipante p = codPart.get(cod);
					Partecipazione part = new Partecipazione(p, reg);
					c.addPart(part);
					line = br.readLine();
				}
				line = br.readLine();
			}
			br.close();
		} catch (IOException e) {
			System.err.println(e);
		}

		// PUNTO 3: stampa di tutti i partecipanti
		System.out.println(
				"tipo, codice, nome, cognome, corso di studi, università, anno di corso, ente di appartenenza, numero di dipendenti, accademico");
		for (Partecipante p : partecipanti)
			System.out.println(p);
		System.out.println();
		// PUNTO 4: stampa conferenze
		for (Conferenza c : conferenze)
			System.out.println(c);

		// PUNTO 5: stampe del totale dei noleggi
		System.out.println();
		String line;
		int cod;
		try {
			BufferedReader kbd = new BufferedReader(new InputStreamReader(System.in));
			line = kbd.readLine();
			cod = Integer.parseInt(line);

			Partecipante part = codPart.get(cod);
			if (part.getTipo().equals("studente")) {
				Studente s = (Studente) part;
				System.out.println("Studente, università: " + s.getUni());
			} else {
				Regolare r = (Regolare) part;
				System.out.println("Regolare, ente: " + r.getEnte());

			}
		} catch (IOException e) {
			System.err.println(e);
		}
	}
}

File: 2-second-year-2022-2023/Programming-languages-and-laboratory/Python/Exercises/26_compito_java_20181119/GestioneConferenze.py
from Partecipante import Partecipante
from Studente import Studente
from Regolare import Regolare
from Conferenza import Conferenza
from Partecipazione import Partecipazione

import sys


class GestioneConferenze:

    def main():

        partecipanti = []
        conferenze = []
        cod_to_partecipante = {}

        try:
            f = open("partecipanti.txt", "r")
            line = f.readline().strip()
            
            while line != "":
                tok = line.split()
                codice = int(tok[0])
                tipo = tok[1]
                nome = f.readline().strip()
                cognome = f.readline().strip()
                corso_ente = f.readline().strip()

                if tipo == "studente":
                    universita = f.readline().strip()
                    anno = int(f.readline().strip())

                    st = Studente(codice, tipo, nome, cognome, corso_ente, universita, anno)
                    partecipanti.append(st)
                    cod_to_partecipante[codice] = st

                else:
                    line = f.readline().strip()
                    tok = line.split()

                    dipendenti = int(tok[0])
                    accademico = bool(tok[1])

                    reg = Regolare(codice, tipo, nome, cognome, corso_ente, dipendenti, accademico)
                    partecipanti.append(reg)
                    cod_to_partecipante[codice] = reg

                
                line = f.readline().strip()

        except IOError as i:
            print(i)
        except Exception as e:
            print(e)

        print("tipo, codice, nome, cognome, corso di studi, università, anno di corso, ente di appartenenza, numero di dipendenti, accademico")

        for p in partecipanti:
            print(p)
        print()

        try:
            f = open("conferenze.txt", "r")
            line = f.readline().strip()

            while line != "":
                nome_conferenza = line
                luogo_conferenza = f.readline().strip()

                conf = Conferenza(nome_conferenza, luogo_conferenza)
                conferenze.append(conf)

                line = f.readline().strip()

                while line != "":
                    tok = line.split()
                    codice = int(tok[0])
                    spesa  = int(tok[1])

                    part = Partecipazione(codice, spesa, cod_to_partecipante[codice].get_cognome())

                    conf.add_partecipazione(part)
                    conf.add_costo(spesa)
                    
                    line = f.readline().strip()

                line = f.readline().strip()
        
        except IOError as i:
            print(i)
        except Exception as e:
            print(e)

        for c in conferenze:
            print(c)
        print()

        try:
            codice_letto = int(sys.argv[1])

            for par in partecipanti:
                if codice_letto == par.get_codice():
                    if par.get_tipo() == "studente":
                        print(str(codice_letto) + "\nStudente, universita: " + par.get_universita())
                    else:
                        print(str(codice_letto) + "\nRegolare, ente: " + par.get_ente())

        except IOError as i:
            print(i)
        except Exception as e:
            print(e)


if __name__ == "__main__":
    GestioneConferenze.main()
